src/marqo/vespa/vespa_client.py

```diff
@@ -675,57 +675,88 @@ async def _update_document_async(self, semaphore: asyncio.Semaphore, async_clien
         doc_id = document.id
         data = {'fields': document.fields}
 
+        # only used for documents that are not updated
+        error_doc_path_id = f"/document/v1/{schema}/{schema}/docid/{doc_id}"
+
         async with semaphore:
             end_point = f'{self.document_url}/document/v1/{schema}/{schema}/docid/{doc_id}?create=false'
             data["condition"] = f'{schema}.{vespa_id_field}==\"{doc_id}\"'
             try:
                 resp = await async_client.put(end_point, json=data, timeout=timeout)
+                return UpdateDocumentResponse(**resp.json(), status=resp.status_code)
             except httpx.HTTPError as e:
-                raise VespaError(e) from e
-
-        try:
-            # This will cover 200 and document-specific errors. Other unexpected errors will be raised.
-            return UpdateDocumentResponse(**resp.json(), status=resp.status_code)
-        except JSONDecodeError:
-            if resp.status_code == 200:
-                # A 200 response shouldn't reach here
-                raise VespaError(f'Unexpected response: {resp.text}')
-
-            self._raise_for_status(resp)
+                error_message = (f"Marqo has encountered an error while updating this document to Vespa and "
+                                 f"the document is not updated. Original error: {e}. "
+                                 f"This might be a transient network issue and you can "
+                                 f"retry the operation. If the issue persists, please contact Marqo support")
+                return UpdateDocumentResponse(status=500, message=error_message, pathId=error_doc_path_id, id=doc_id)
+            except JSONDecodeError as e:
+                if resp.status_code == 200:
+                    # A 200 response shouldn't reach here
+                    raise ValueError(f"Marqo received an unexpected response from Vespa: {resp.text} with 200 "
```

Comment:
I feel like it's okay to fail the whole batch and raise a 500 in this case because we don't really know what we're dealing with when there's a `JSONDecodeError` and it's not a network issue. Keen to get @papa99do 's review here